connectors/sources/tests/test_confluence.py

```diff
@@ -0,0 +1,151 @@
+
+    # Setup
+    source = create_source(ConfluenceDataSource)
+    options = {"concurrent_downloads": 10}
+
+    # Execute
+    source.tweak_bulk_options(options)
+
+
+@pytest.mark.asyncio
+async def test_close_with_client_session(patch_logger):
+    """Test close method for closing the existing session"""
+
+    # Setup
+    source = create_source(ConfluenceDataSource)
+    source._generate_session()
+
+    # Execute
+    await source.close()
+
+    # Assert
+    assert source.session is None
+
+
+@pytest.mark.asyncio
+async def test_close_without_client_session(patch_logger):
+    """Test close method when the session does not exist"""
+    # Setup
+    source = create_source(ConfluenceDataSource)
+
+    # Execute
+    await source.close()
+
+    # Assert
+    assert source.session is None
+
+
+class MockSSL:
+    """This class contains methods which returns dummy ssl context"""
+
+    def load_verify_locations(self, cadata):
+        """This method verify locations"""
+        pass
+
+
+@pytest.mark.asyncio
+async def test_configuration():
+    """Tests the get_default_configurations method of the Confluence source class."""
+    # Setup
+    klass = ConfluenceDataSource
+
+    # Execute
+    config = DataSourceConfiguration(config=klass.get_default_configuration())
+
+    # Assert
+    assert config["host_url"] == HOST_URL
+
+
+@pytest.mark.asyncio
+async def test_validate_configuration_for_host_url(patch_logger):
+    """This function test _validate_configuration when host_url is invalid"""
+    # Setup
+    source = create_source(ConfluenceDataSource)
+    source.configuration.set_field(name="host_url", value="")
+
+    # Execute
+    with pytest.raises(Exception):
+        await source.validate_config()
+
+
+@pytest.mark.asyncio
+@patch("aiohttp.ClientSession.get")
+async def test_ping_with_ssl(mock_get, patch_logger):
+    """Test ping method of ConfluenceDataSource class with SSL"""
+
+    # Execute
+    mock_get.return_value.__aenter__.return_value.status = 200
+    source = create_source(ConfluenceDataSource)
+
+    source.ssl_enabled = True
+    source.certificate = (
+        "-----BEGIN CERTIFICATE----- Certificate -----END CERTIFICATE-----"
+    )
+
+    # Execute
+    with patch.object(ssl, "create_default_context", return_value=MockSSL()):
+        source.ssl_ctx = ssl_context(certificate=source.certificate)
+        await source.ping()
+
+
+@pytest.mark.asyncio
+@patch("aiohttp.ClientSession.get")
+async def test_ping_for_failed_connection_exception(mock_get, patch_logger):
+    """Tests the ping functionality when connection can not be established to Confluence."""
+
+    # Setup
+    source = create_source(ConfluenceDataSource)
+
+    # Execute
+    with patch.object(
+        ConfluenceDataSource, "_api_call", side_effect=Exception("Something went wrong")
+    ):
+        with pytest.raises(Exception):
+            await source.ping()
+
+
+def test_validate_configuration_for_ssl_enabled(patch_logger):
```

Comment:
We may want to have a test also validating that the fields are non-empty.

Reply:
This already exists as `test_validate_configuration_for_host_url`

Reply:
But not for "service_account_id", "api_token", if it's cloud and "username", "password", if it's Jira server, right? You can use `pytest.parametrize` to reuse the testing logic.